### DISCUSSION TITLE
Serialize chunks back to .md files for backup

### DISCUSSION THREAD

**Discussion by MattSzymonski** [vrtmrz/obsidian-livesync#313] [General]
Hi, LiveSync is absolutely great. Thank you and keep up the good job!

I like the chunking idea however, I am afraid of storing my notes in this form.
Mostly because I can't serialize them back to .md files without using LiveSync.

I'm hosting CouchDB on my server and I would like to perform a backup copy from time to time.
But not an exact copy of the database. I want to convert everything back to proper files and back it up like this.

I can create a cron script that will take care of this, but I need a way (some code/program/api) to perform this chunk serialization/conversion.

Is there already some way to do this? If not, are there any chances for something like this to appear?

**Comment by vrtmrz** [comment]
Thank you for asking me!
We can manipulate the files with [DirectFileManipulator.ts](https://github.com/vrtmrz/livesync-commonlib/blob/main/src/DirectFileManipulator.ts) 
I was just reworking the integration of [Filesystem-LiveSync](https://github.com/vrtmrz/filesystem-livesync) and [LiveSync-Classroom](https://github.com/vrtmrz/livesync-classroom), but I had been hooked a bit of interruptive work while that. However, if you are familiar with TypeScript, you could use it before my release! (Sorry for lacking the documents).
This is already used by [LiveSync-Subscribe-Publish](https://github.com/vrtmrz/livesync-subscribe-publish).

**Reply by MattSzymonski** [reply]
Thank you for the quick answer!
